src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -124,37 +128,35 @@ public void cancelOperation_usingAccelerator() {
-        editPersonDialog.enterNewValues(newAlice);
-        editPersonDialog.pressEnter();
+        EditPersonDialogHandle editPersonDialog = personListPanel.editPerson(td.alice);
+        editPersonDialog.enterNewValues(newAlice).pressEnter();
 
+        //Ensure grace period is showing
+        assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
+
+        //Edit Alice again during pending state.
         personListPanel.clickOnPerson(newAlice);
-        assertEquals(alicePersonCard, newAlice);
-        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
-        assertEquals(alicePersonCard, td.alice);
-        assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] was cancelled.");
+        Person newerAlice = new PersonBuilder(newAlice.copy()).withFirstName("Felicia").withLastName("Yellowstone")
+                .withStreet("street updated").withCity("Malaysia").withPostalCode("321321")
+                .withBirthday("11.11.1979").withGithubUsername("yellowstone").withTags(td.colleagues).build();
 
-        //New
-        EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
-        Person pandaWong = new PersonBuilder("Panda", "Wong")
-                .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
-                .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
-        addPersonDialog.enterNewValues(pandaWong);
-        addPersonDialog.clickOk();
+        editPersonDialog = personListPanel.editPerson(newAlice);
 
-        personListPanel.clickOnListView(); // To ensure shortcut keys work properly.
+        //Get a reference to the current
 
-        personListPanel.use_LIST_GOTO_BOTTOM_SEQUENCE();
-        PersonCardHandle pandaWongCardHandle = personListPanel.getPersonCardHandle(pandaWong);
-        assertEquals(pandaWongCardHandle, pandaWong);
-        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
-        assertNull(personListPanel.getPersonCardHandle(pandaWong));
+        //Ensure grace period is not counting down while editing person.
+        assertTrue(alicePersonCard.isGracePeriodFrozen());
+
+        editPersonDialog.enterNewValues(newerAlice).pressEnter();
 
-        //Ensure cancel operation has no side effects after grace period.
+        //Ensure card is displaying Felicia before and after grace period.
+        assertMatching(alicePersonCard, newerAlice);
         sleepForGracePeriod();
-        assertNull(personListPanel.getPersonCardHandle(pandaWong));
-        personListPanel.isListMatching(td.alice, td.benson, td.charlie, td.dan, td.elizabeth);
+        assertTrue(personListPanel.isListMatching(newerAlice, td.benson, td.charlie, td.dan, td.elizabeth));
     }
 
-    //TODO: testing edits during grace period
+    @Test
+    public void cancelOperation_afterGracePeriod() {
+        //Tested in editPerson_usingContextMenu()
+    }
```

Comment:
Did you include a test to confirm the TAB order is working correctly for the edit dialog?